Makefile

```diff
@@ -1,10 +1,7 @@
-# get Makefile directory name: http://stackoverflow.com/a/5982798/376773
-THIS_MAKEFILE_PATH:=$(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
-THIS_DIR:=$(shell cd $(dir $(THIS_MAKEFILE_PATH));pwd)
+SINGLE_QUOTE := '
+# We need to escape single quote as '\''
+THIS_DIR := '$(subst $(SINGLE_QUOTE),$(SINGLE_QUOTE)\$(SINGLE_QUOTE)$(SINGLE_QUOTE),$(CURDIR))'
```

Comment:
`CURDIR` is a variable set by `make` [pointing to the current working directory](https://www.gnu.org/software/make/manual/html_node/Recursion.html#index-CURDIR).
I don't see much sense in jumping through hoops to get the CWD, when we don't actually support invoking `make` for any other directory than the root (where the `Makefile` resides).